src/main/java/io/rcktapp/api/handler/firehose/FirehoseDb.java

```diff
@@ -94,28 +94,26 @@ public void bootstrapApi() throws Exception {
             collection.setEntity(entity);
 
             api.addCollection(collection);
-        }
+        });
     }
 
-    public AmazonKinesisFirehoseAsync getFirehoseClient() {
-        if (this.firehoseClient == null) {
-            synchronized (this) {
-                if (this.firehoseClient == null) {
-                    AmazonKinesisFirehoseAsyncClientBuilder builder = AmazonKinesisFirehoseAsyncClientBuilder.standard();
-                    if (!J.empty(awsRegion))
-                        builder.withRegion(awsRegion);
-
-                    if (!J.empty(awsAccessKey) && !J.empty(awsSecretKey)) {
-                        BasicAWSCredentials creds = new BasicAWSCredentials(awsAccessKey, awsSecretKey);
-                        builder.withCredentials(new AWSStaticCredentialsProvider(creds));
-                    }
-
-                    firehoseClient = builder.build();
-                }
-            }
+    public synchronized AmazonKinesisFirehoseAsync getFirehoseClient() {
+        if (this.firehoseClient != null)
+            return this.firehoseClient;
```

Comment:
If you do this pattern (return early if you can) the code is faster to read when you know what the states are - you don't have to go on safari through whatever code to find out what the default is!  There's a method to this madness.

Also, the code below is not indented so many times any more!  Yay!